### DISCUSSION TITLE
发生异常: TypeLoadException 未能从程序集“System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a”中加载类型“System.Environment”。

### DISCUSSION THREAD

**Discussion by heartacker** [pythonnet/pythonnet#1827] [Q&A]
### Environment

-   Pythonnet version: https://pypi.org/project/pythonnet/3.0.0a2/
-   Python version: 3.9
-   Operating System: win 11
-   .NET Runtime: 6.0.301

### Details

-   Describe what you were trying to get done.

    I had developed a c# dll using .NET 6, and I try to use pythonnet clr to load those assemely.
    those is the related info:
1. csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Library</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
    <Platforms>AnyCPU;x64</Platforms>
  </PropertyGroup>

```
3. csharp code
```csharp
using Env = System.Environment;
...

        public static void Prefer(ChipType chipType)
        {
            if (Env.GetEnvironmentVariables().Contains("PYTHON_IN_DOTNET"))// running code Studio
            {
                PreferedChipType = chipType;
            }
        }
```

5. errmsg when invoke
```
发生异常: TypeLoadException
未能从程序集“System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a”中加载类型“System.Environment”。
   在 ....Prefer(ChipType chipType)
```
![image](https://user-images.githubusercontent.com/1876302/174986544-04167095-0406-4b45-a30e-65ae03a188a6.png)

the msg shown that the runtime is `Version=4.0.0.0`, not 6.0.0
#region 程序集 System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
// System.Runtime.dll

**Comment by iamtorchstack** [comment]
Hi Hearttacker, I had similar issue with pythonnet. I reduced the .NetFramework used to generate the DLL file from .Netcore 6.0 to 4.8 and I use the regular command in the pythonnet documentation and it worked.

**Reply by iamtorchstack** [reply]
I want to believe that pythonnet 3.0XX is not yet capable to handle reading DLL generate in more recent .NET frameworks

**Reply by filmor** [reply]
That's not true. Please check the release candidate announcement, you have to explicitly request a .NET Core (`coreclr`) runtime from clr loader. The default is the same as in older versions (so .NET Framework on Windows and Mono on Linux and macOS).

**Comment by heartacker** [comment]
@torch016 I find a workaround to deal this problem.

https://github.com/pythonnet/pythonnet/issues/1634#:~:text=no%20attribute%20%27_handle%27-,My%20Python%20code%3A,-import%20os%0Aimport
we could set the runtime explicitly using runtime.config
![image](https://user-images.githubusercontent.com/1876302/181702048-a8fc55db-eb5b-40b0-a6ea-abd6b183fa7f.png)
